### ISSUE TITLE
Editor: Open the site picker before entering the editor (to match web)

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#6218]
[Raised](https://test.wordpress.com/2016/11/10/session-notes-back-buttons-hidden-sites-touch-image-flow-notifications-site-switching-flow-keyboard-flow-media-section-desktop-first/) by Ryan:

> I almost published to the wrong site twice in my last session. I had to ditch in progress posts with images twice. I wish the iOS site picker had multisite flow more like Calypso’s. It asks which site to publish to on the way into the editor.

In Calypso web, when you tap the editor icon to open the editor, a site picker appears. You can then select the site you want to post to before entering the editor:

![screen shot 2016-11-21 at 4 27 50 pm](https://cloud.githubusercontent.com/assets/8658164/20490996/d00483ee-b007-11e6-8fac-009761140b14.png)

This helps reduce posting anxiety, e.g. fear that you will accidentally post to the wrong site. It also prevents you from starting a post and adding media to the wrong site.

Currently, depending on context the app's editor button may open your primary site, the last site you posted to, or the site you're currently working on. Related issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/5480

It would be interesting to explore a flow more similar to the Calypso flow, which presents the site picker on the way in to the editor.

##### Tested on iPhone 6, iOS 10.1.1, WPiOS 6.7.0.20161115

**Comment by SergioEstevao** [comment]
@astralbodies do you think this is a good change? there has been some conversation around this one...

**Comment by astralbodies** [comment]
I think it's something we need to consider across platforms. I'd suggest bringing this up for discussion amongst the team.

**Comment by bummytime** [comment]
Pinging @mattmiklic and/or @folletto for their input as to whether this issue is still valid.

😄

**Comment by mattmiklic** [comment]
CC:ing @iamthomasbishop here since he's thinking about the publishing flow; this may be food for thought there.

**Comment by folletto** [comment]
Contextual note—

Currently the editor on mobile opens and allows site switching after the editor is opened. Thus the site switching feature is already provided, even if in a different order (and there's a big discussion about defaults here which I won't address in this comment).

The post can be switched of site under a set of conditions:

1. On creation we allow switching.
2. When an image is uploaded it gets in a “semi-locked” state: we warn the user “Choosing a different site will lose edits to site specific content like media and categories. Are you sure?”.
3. Once the post is closed the site can’t be changed anymore when edited again.

As a reference, this was discussed in relation to Gutenberg mobile here: p77Llu-9DQ-p2

**Comment by bummytime** [comment]
@folletto completely agree. 💯‼️

I will wait for @iamthomasbishop’s comment here, but it seems to make sense to close this issue in favor of all of the Gutenberg mobile work.

**Comment by iamthomasbishop** [comment]
I've thought about this a bit in the past month or 2, and while I initially thought it'd be good to match Calypso, I decided to hold off on it for now partially because of the reasons @folletto mentioned above. 

It's also worth noting is that the vast majority of users don't have multiple sites - so while this might be an annoyance for us that have a bunch of sites, most folks don't. I came across data on this recently but I'm unable to find atm. 

Another point is that we have discussed potentially re-working the nav structure/hierarchy in the apps later in the year, so I'm hesitant to make big changes before that.

With all that said, this is something I intend on testing in the first iterations of the mobile Gutenberg proof-of-concept, so my vote is to hold off on changes for right now, and re-assess based on testing feedback.
